Description: Link shared libraries against their direct dependencies
 libar-util uses symbols from libar-osal, while libar-ats uses symbols from
 libar-gsl.  Recording those dependencies at link time gives the libraries
 the required DT_NEEDED entries and prevents unresolved-symbol warnings from
 dpkg-shlibdeps.
Author: Lukas Märdian <lukas.maerdian@oss.qualcomm.com>
Forwarded: https://github.com/AudioReach/audioreach-graphservices/pull/109
Last-Update: 2026-09-01
--- a/ar_util/Makefile.am
+++ b/ar_util/Makefile.am
@@ -23,4 +23,5 @@ lib_LTLIBRARIES = libar-util.la
 libar_util_la_CC = @CC@
 libar_util_la_SOURCES = $(util_c_sources)
 libar_util_la_CFLAGS = $(AM_CFLAGS)
+libar_util_la_LIBADD = $(top_builddir)/ar_osal/libar-osal.la
 libar_util_la_LDFLAGS = -shared -version-number @LT_VERSION_NUMBER@
--- a/acdb/ats/Makefile.am
+++ b/acdb/ats/Makefile.am
@@ -67,8 +67,9 @@ requiredlibs = -lrt
 
 lib_LTLIBRARIES = libar-ats.la
 libar_ats_la_CC = @CC@
-libar_ats_la_LIBADD = -lar-osal -L$(top_builddir)/ar_osal
-libar_ats_la_LIBADD += -lar-acdb -L$(top_builddir)/acdb
+libar_ats_la_LIBADD = $(top_builddir)/ar_osal/libar-osal.la
+libar_ats_la_LIBADD += $(top_builddir)/acdb/libar-acdb.la
+libar_ats_la_LIBADD += $(top_builddir)/gsl/libar-gsl.la
 if CUTILS_SUPPORTED
 libar_ats_la_LIBADD += -lcutils
 AM_CFLAGS += -DAR_ATS_USE_CUTILS
